sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py

```diff
@@ -236,11 +251,12 @@ def create_table(cls, table_name):
     cls.bigquery_client.insert_rows(
         cls.project, cls.dataset_id, table_name, table_data)
 
-  def get_expected_data(self):
+  def get_expected_data(self, native=True):
+    byts = b'\xab\xac'
     expected_row = {
         'float': 0.33,
         'numeric': Decimal('10'),
-        'bytes': base64.b64encode(b'\xab\xac'),
+        'bytes': base64.b64encode(byts) if native else byts,
```

Comment:
How intuitive will this behavior be to users? should't the output be consistent? 
If we decide to keep it, how will this be documented?

Reply:
The behavior will be different for different transforms. Users will need to explicitly change the transform in their code. We can make them aware of the new transform, and its typing differences in release notes, and possibly in Pydoc for the new transform as well. Thoughts?

Reply:
Bytes treatment should be called out in IO doc, we do mention it: https://github.com/apache/beam/blob/8bc2880cca40c00a96623b3ce96ea0b856af76c9/sdks/python/apache_beam/io/gcp/bigquery.py#L227. 

b64encoding may be unnecessary, and less efficient. I think the reason native IO encodes bytes with b64 is because that was the behavior in Java SDK. We can argue that's not necessary. However I am concerned about the consistency of the UX here. Different UX for two transforms means the transforms will not be interchangeable, and users might overlook this.  This might also cause friction in cross-language pipelines.

Reply:
I think that's reasonable. But I'd think that after slight confusion, users would figure out the differences.

Another option is to provide a parameter to the transform to have backwards-compatible types at a performance cost.

@chamikaramj @tvalentyn  wdyt?

Reply:
Having a parameter for backwards-compatibility would make sense to me, but will defer to @chamikaramj  on this. Since new transform comes with a new name, whoever will be using it, will have to read the API, so will be aware of the issue. But that would mean we add a knob, potentially indefinitely.

Reply:
So the proposal right now is: To have a parameter for backwards compatibility in the transform, but by default return Avro-returned types. We will need to have clear Pydoc on _ReadFromBigQuery, and on CHANGES.md.
@chamikaramj thoughts?

Reply:
@chamikaramj @tvalentyn @kamilwu - added a backward compatibility flag. Can you review?

Reply:
If the changes are reasonable, I'll add to release notes

Reply:
I think having a single "backwards compatibility" parameter can be confusing and error prone. We might still be backwards incompatible in some unknown ways. I think it's better to document exact differences between the transforms and give clear documentation/guidelines for anyone who is upgrading.